test/join.cpp

```diff
@@ -44,7 +44,7 @@ class Join : public ::testing::Test {
 
 // create a list of types to be tested
 typedef ::testing::Types<float, double, cfloat, cdouble, int, unsigned int,
-                         intl, uintl, char, unsigned char, short, ushort>
+                         intl, uintl, char, unsigned char, short, ushort, af_half>
```

Comment:
we should be C++ typedef `af::half` to be consistent with `cfloat` and `cboudle`. This applies to all the tests where `half` is enabled.

Reply:
I am worried we will have conflicts with the half data type. I want to discourage the use of this type by users because of its inefficiency. I am not sure if this is the best way to handle this sort of thing.